### PR TITLE
Add s2wasm option to ignore unknown symbols

### PR DESCRIPTION
**Pull request by jfbastien**
This is useful for bringup.

### CODE REVIEW

File: src/s2wasm.h [hunk 1]
```diff
@@ -38,10 +38,17 @@ class S2WasmBuilder {
...
   bool debug;
-
-public:
- S2WasmBuilder(AllocatingModule& wasm, const char* input, bool debug, size_t globalBase)
-     : wasm(wasm), allocator(wasm.allocator), debug(debug), globalBase(globalBase), nextStatic(globalBase) {
+  bool ignore_unknown_symbols;
+
+ public:
+  S2WasmBuilder(AllocatingModule& wasm, const char* input, bool debug,
+                size_t globalBase, bool ignore_unknown_symbols)
```
Comment (kripken) [inline src/s2wasm.h]: should use camelCase like in the argument right before it

File: src/s2wasm.h [hunk 2]
```diff
@@ -50,7 +57,7 @@ class S2WasmBuilder {
     fix();
   }
 
-private:
+ private:
```
Comment (kripken) [inline src/s2wasm.h]: eh?

**Comment by kripken** [comment]
lgtm otherwise. please rebase before merge.

File: src/s2wasm.h (hunk 2 above)
Comment (jfbastien) [inline src/s2wasm.h]: clang-format, it sometimes catches the next line and formats it too.

**Comment by jfbastien** [comment]
@kripken do you know what this vanilla LLVM error is?

File: src/s2wasm.h (hunk 2 above)
Comment (kripken) [inline src/s2wasm.h]: oh ok.

**Comment by kripken** [comment]
Link?

**Comment by jfbastien** [comment]
Travis: https://travis-ci.org/WebAssembly/binaryen/jobs/105331090

**Comment by kripken** [comment]
That doesn't look like a vanilla LLVM error? It's failing to run `asm2wasm`, because CMake failed so nothing could be built (and `asm2wasm` is just the first it tries),

````
$ cmake . -DCMAKE_C_FLAGS="$COMPILER_FLAGS" -DCMAKE_CXX_FLAGS="$COMPILER_FLAGS"

CMake Error at /usr/share/cmake-2.8/Modules/CMakeDetermineCXXCompiler.cmake:43 (MESSAGE):

  Could not find compiler set in environment variable CXX:

  clang++-3.6.
````

It seems to happen randomly now and then. Some travis bots are just bad I guess?

**Comment by jfbastien** [comment]
Ah OK, should I just merge then?

**Comment by kripken** [comment]
Yes, but please rebase/squash.

**Comment by jfbastien** [comment]
Waiting on branch alias: I made this PR based on it because GH was down :(
I'm sure there's a way to fix it, but I'm sure it's easy to mess up.

**Comment by kripken** [comment]
I don't think you can redirect a pull on github. Just rebase locally to master and merge.

**Comment by jfbastien** [comment]
Yeah I messed things up... Not a big issue, leaving as-is so Dan can chime in on #148.

**Pull request closed without merging**